pkg/controllers/management/drivers/nodedriver/machine_driver.go

```diff
@@ -136,6 +166,35 @@ func (m *Lifecycle) download(obj *v3.NodeDriver) (*v3.NodeDriver, error) {
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return obj, err
 	}
+	obj, err = m.createCredSchema(obj, credFields)
+	if err != nil {
+		return obj, err
+	}
+
+	//passwordStore.SetPasswordStore(m.schemas, m.secretStore, m.nsStore)
+
+	return obj, nil
+}
+
+func (m *Lifecycle) createCredSchema(obj *v3.NodeDriver, credFields map[string]v3.Field) (*v3.NodeDriver, error) {
+	credentialSchema := &v3.DynamicSchema{
+		Spec: v3.DynamicSchemaSpec{
+			ResourceFields: credFields,
+		},
+	}
+	credentialSchema.Name = obj.Spec.DisplayName + "credentialconfig"
+	credentialSchema.OwnerReferences = []metav1.OwnerReference{
+		{
+			UID:        obj.UID,
+			Kind:       obj.Kind,
+			APIVersion: obj.APIVersion,
+			Name:       obj.Name,
+		},
+	}
+	_, err := m.schemaClient.Create(credentialSchema)
```

Comment:
is that easier to always create it, and check for the error, vs fetching the object first to check if it exists?